pyg_lib/csrc/sampler/cpu/random_walk_kernel.cpp

```diff
@@ -0,0 +1,53 @@
+#include <torch/torch.h>
+
+#include "utils.h"
+
+namespace pyg {
+namespace sampler {
+
+namespace {
+
+torch::Tensor random_walk_kernel(const torch::Tensor& rowptr,
+                                 const torch::Tensor& col,
+                                 const torch::Tensor& seed,
+                                 int64_t walk_length,
+                                 double p,
+                                 double q) {
+  TORCH_CHECK(p == 1 && q == 1, "Uniform sampling required for now");
+
+  const auto out = rowptr.new_empty({seed.size(0), walk_length + 1});
+
+  AT_DISPATCH_INTEGRAL_TYPES(rowptr.scalar_type(), "random_walk_kernel", [&] {
+    const auto rowptr_data = rowptr.data_ptr<scalar_t>();
+    const auto col_data = col.data_ptr<scalar_t>();
+    const auto seed_data = seed.data_ptr<scalar_t>();
+    auto out_data = out.data_ptr<scalar_t>();
+
+    auto grain_size = at::internal::GRAIN_SIZE / walk_length;
+    at::parallel_for(0, seed.size(0), grain_size, [&](int64_t _s, int64_t _e) {
+      for (auto i = _s; i < _e; i++) {
+        scalar_t v = seed_data[i], row_start, row_end, rand;
```

Comment:
```suggestion
        scalar_t v = seed_data[i], row_start, row_end, rand;
```
```suggestion
        scalar_t v = seed_data[i];
```

Are the last three variables not being used?

Reply:
They are used in the inner loop. No preference though.

Reply:
In google style guide, it is always recommend to put this type of light variable together with initialization. I'd suggest move them in the loop. Also easy to read.